Client/AdjustTelemetryHelper.swift

```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Adjust
+import Glean
+
+protocol AdjustTelemetryData {
+    var campaign: String? { get set }
+    var adgroup: String? { get set }
+    var creative: String? { get set }
+    var network: String? { get set }
+}
+
+extension ADJAttribution: AdjustTelemetryData {
+}
+
+protocol AdjustTelemetryProtocol {
+    func setAttributionData(_ attribution: AdjustTelemetryData?) -> Bool
+    func sendDeeplinkTelemetry(url: URL, attribution: AdjustTelemetryData?)
+}
+
+class AdjustTelemetryHelper: AdjustTelemetryProtocol {
+
+    func sendDeeplinkTelemetry(url: URL, attribution: AdjustTelemetryData?) {
+        let extra = GleanMetrics.Adjust.DeeplinkReceivedExtra(receivedUrl: url.absoluteString)
+        GleanMetrics.Adjust.deeplinkReceived.record(extra)
+
+        _ = setAttributionData(attribution)
```

Comment:
MInor comment, for testing we could avoid returning a true/false and just rely on verifying the set worked by checking Glean has the value recorded or not. Would to clarify the AdjustTelemetryProtocol interface